halo/evmstaking2/keeper/keeper.go

```diff
@@ -37,12 +40,38 @@ func NewKeeper(storeService store.KVStoreService) (*Keeper, error) {
 	}
 
 	return &Keeper{
-		eventsTable: evmstakingStore.EVMEventTable(),
+		eventsTable:     evmstakingStore.EVMEventTable(),
+		submissionDelay: submissionDelay,
 	}, nil
 }
 
-func (*Keeper) EndBlock(context.Context) ([]abci.ValidatorUpdate, error) {
-	return nil, nil
+// EndBlock delivers all pending EVM events on every `k.submissionDelay`'th block.
+func (k *Keeper) EndBlock(ctx context.Context) error {
+	blockHeight := sdk.UnwrapSDKContext(ctx).BlockHeight()
+
+	if blockHeight%k.submissionDelay != 0 {
+		return nil
+	}
+
+	eventIter, err := k.eventsTable.List(ctx, EVMEventIdIndexKey{})
```

Comment:
nit: can just name it `iter`, there are no other iters to confuse it with. Concise naming FTW 🚀